snappiershot/compare.py

```diff
@@ -66,6 +76,25 @@ def _compare(self, value: Any, expected: Any, *, operations: List[Callable]) ->
         if value != expected:
             return self.differences.add(operations, f"{value} != {expected}")
 
+    def _compare_units(
+        self, value: Unit, expected: Unit, *, operations: List[Callable] = None
+    ) -> None:
+        """ Perform a recursive, almost-equals comparison between value and expected.
+
+        This is a helper function for when the expected is a unit, it compares objects without checking type.
+
+        Args:
+            value: The object to be checked.
+            expected: The object to be compared against.
+            operations: **Internally used for recursion**
+              Tracks the operations that need to be applied to self.value and self.expected
+                to obtain value and expected, respectively. Used for logging differences.
+        """
+        # Check if the string format of both are equal due to possibility of different registries
+        if value.__str__() != expected.__str__():
```

Comment:
Could we do `str(value) != str(expected)`?

Reply:
[fixed](https://github.com/MORSECorp/snappiershot/pull/96/commits/a15fc65460a35c3f3f31dd9ca92f52e4a0684fe1)